svm/src/rollback_accounts.rs

```diff
@@ -51,6 +51,10 @@ impl RollbackAccounts {
 
         if let Some(nonce) = nonce {
             if &fee_payer_address == nonce.address() {
+                fee_payer_account
+                    .data_as_mut_slice()
+                    .copy_from_slice(nonce.account().data());
```

Comment:
please let me know if there is a more idiomatic/less ugly way to reassign data for `AccountSharedData`

Reply:
We shouldn't need to update the account data here right? The fee payer account data is already the same as the nonce account data because they are the same account.

Reply:
we do need to, for the `SameNonceAndFeePayer` case. theres two copies of the data, the fee-payer account (FP) and nonce info (NI)

the way it works in master is fees and rent are charged on FP in `validate_transaction_fee_payer`. NI isnt mutated. `RollbackAccounts::new()` refunds rent on FP, checks if NI exists, and if it does creates a new nonce info from FP. the nonce is advanced in account-saver

in this pr, fees and rent are charged on FP in `validate_transaction_fee_payer`. NI is advanced to the next nonce. `RollbackAccounts::new()` refunds rent on FP and then if NI exists copies its data to FP; before this, NI has the data we want and FP has the lamports and rent epoch we want

Reply:
ah yeah got it, thanks. Maybe it's cleaner to move this inside `RollbackAccounts::new`?

Reply:
i changed this to the `AccountSharedData` method `set_data_from_slice()`, which seems more polite than the previous. ultimately it needs to be copied here unless we wrap "nonce and fee payer" in one object (which i dont think it worth it) because it doesnt look like `AccountSharedData` has a mechanism to just reassign the `Arc` from one to the other, and the place in `RollbackAccounts::new()` where this happens is the right spot to resolve the "same or different accounts" question